File: modules/core-api/src/main/java/com/enonic/xp/schema/content/ContentTypesDeletedEvent.java
package com.enonic.xp.schema.content;

import java.util.ArrayList;
import java.util.List;

import com.enonic.xp.event.Event;

public class ContentTypesDeletedEvent
    implements Event
{
    private List<ContentTypeName> names;

    public ContentTypesDeletedEvent( )
    {
        names = new ArrayList<ContentTypeName>();
    }

    public void addContentTypeName( final ContentTypeName name ) {
        names.add( name );
    }

    public List<ContentTypeName> getNames()
    {
        return names;
    }
}

File: modules/admin-ui/src/main/resources/web/admin/common/js/schema/content/ContentTypesDeletedEvent.ts
module api.schema.content {

    export class ContentTypesDeletedEvent extends api.event.Event {

        private contentTypeNames: ContentTypeName[];

        constructor(names: string[]) {
            super();
            this.contentTypeNames = names.map((name: string) => {
                return new ContentTypeName(name);
            });
        }

        getContentTypeNames(): ContentTypeName[] {
            return this.contentTypeNames;
        }

        static on(handler: (event: ContentTypesDeletedEvent) => void) {
            api.event.Event.bind(api.ClassHelper.getFullName(this), handler);
        }

        static un(handler?: (event: ContentTypesDeletedEvent) => void) {
            api.event.Event.unbind(api.ClassHelper.getFullName(this), handler);
        }


        static fromJson(json: ContentTypeDeletedEventJson): ContentTypesDeletedEvent {
            return new ContentTypesDeletedEvent(json.names);
        }
    }

    export interface ContentTypeDeletedEventJson {
        names: string[];
    }
}